### PR TITLE
Issues/3338 use natural image size

### LINKED ISSUE #3338: Use natural size when calculating new images sizes, and constrain to max size.
Title has it.

### PR DESCRIPTION
**Pull request by aerych**
Closes #3338 

Improves image resize behavior in the edit image modal.

### CODE REVIEW

**Comment by aerych** [comment]
cc @bummytime for review por favor :)

File: Podfile [hunk 1]
```diff
@@ -27,7 +27,7 @@ pod 'HockeySDK', '~>3.6.0'
 pod 'Helpshift', '~>4.8.0'
 pod 'CTAssetsPickerController', '~> 2.7.0'
 pod 'WordPress-iOS-Shared', '0.2.2'
-pod 'WordPress-iOS-Editor', :git => 'git://github.com/wordpress-mobile/WordPress-Editor-iOS', :commit => 'bd67b336dd1ca5bc2c5a277fa08f3dac017adb17'
+pod 'WordPress-iOS-Editor', :git => 'git://github.com/wordpress-mobile/WordPress-Editor-iOS', :commit => '6882a21fc71d6c02efef277a5178f7c1f82121a6'
```
Comment (bummytime) [inline Podfile]: Should we update this SHA to ```27089b8d74efc07e2743de924edd47989a4456d9```?

File: Podfile.lock [hunk 2]
```diff
@@ -150,7 +150,7 @@ DEPENDENCIES:
   - SVProgressHUD (from `https://github.com/TransitApp/SVProgressHUD.git`, commit `2ac46ae2f2dd37db153ce1daea73b1273ac2bba3`)
   - UIDeviceIdentifier (~> 0.1)
   - WordPress-AppbotX (from `https://github.com/wordpress-mobile/appbotx.git`, commit `303b8068530389ea87afde38b77466d685fe3210`)
-  - WordPress-iOS-Editor (from `git://github.com/wordpress-mobile/WordPress-Editor-iOS`, commit `bd67b336dd1ca5bc2c5a277fa08f3dac017adb17`)
+  - WordPress-iOS-Editor (from `git://github.com/wordpress-mobile/WordPress-Editor-iOS`, commit `6882a21fc71d6c02efef277a5178f7c1f82121a6`)
```
Comment (bummytime) [inline Podfile.lock]: Should we update this SHA to ```27089b8d74efc07e2743de924edd47989a4456d9```?

File: Podfile (hunk 1 above)
Comment (aerych) [inline Podfile]: Totes! That one didn't exist when I made the PR. Now that the pod is updated we can point to the latest and greatest.

**Comment by bummytime** [comment]
Sorry this took so long @aerych...I got distracted ;-)  This fix tested well and hopefully we have all the issues with the resize behavior. After updating the SHA, I say :shipit:

**Comment by aerych** [comment]
Thanks @bummytime!

**Pull request merged**